Add transcoding of api sample hlsl_shaders based on vulkan.hpp

asuessenbach opened the pull request:
## Description

Introduces a new Vulkan-Hpp-based sample, which is a transcoded version of the hlsl_shaders sample.

Build tested on Windows with NVIDIA GPU.

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [x] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

asuessenbach committed:
Change vkb::core::HPPPhysicalDevice and vkb::core::HPPInstance from facades over vkb::PhysicalDevice and vkb::Instance, respectively, to self-contained classes using vulkan.hpp

asuessenbach committed:
Explicitly assign HPPStructurType::structureType to a variable to be usable in std::map::find().

asuessenbach committed:
Change vkb::core::HPPVulkanResource from a facade over vkb::core::VulkanResource to a self-contained class using vulkan.hpp

asuessenbach committed:
Add transcoding of api sample terrain_tessellation based on vulkan.hpp

asuessenbach committed:
Change vkb::core::HPPQueue from a facade over vkb::Queue to a self-contained class using vulkan.hpp

asuessenbach committed:
Add transcoding of api sample hlsl_shaders based on vulkan.hpp

SaschaWillems reviewed (commented):
The sample itself looks good to me and works for me on Windows 10 (RTX 2060) and Android (Google Pixel 4a).

The [CMakeLists.txt](samples/api/hpp_hlsl_shaders/CMakeLists.txt) for the sample needs to be adjusted though. It's a 1:1 copy of the non-HPP sample. As the Android launcher uses the info from this file for the sample list, you currently can't tell which of the HLSL samples is the HPP one:

![image](https://user-images.githubusercontent.com/10283127/183705015-9e3f2ead-b4bb-4551-ab46-14e069e9d3f5.png)

Other than that all changes LGTM.

asuessenbach committed:
Prepend "HPP" in NAME in CMakeLists.txt

plus modified hpp_hello_triangle/CMakeLists.txt accordingly to have a consistent naming scheme.
